crates/driver/src/simulator/ethereum.rs

```diff
@@ -1,10 +1,14 @@
 use crate::logic::eth;
 
 #[derive(Debug)]
-pub(super) struct Ethereum;
+pub(super) struct Ethereum(crate::Ethereum);
 
 impl Ethereum {
-    pub async fn simulate(&self, _tx: &eth::Tx, _access_list: &eth::AccessList) -> eth::Simulation {
+    pub async fn simulate(
+        &self,
+        _tx: &eth::Tx,
+        _access_list: &eth::AccessList,
+    ) -> super::Simulation {
```

Comment:
As mentioned, I don't see a reason for access list to be a special parameter here. It's just another tx parameter as all others.